### PR TITLE
feat: Add structured data for partners GRO-379

### PR DESCRIPTION
**Pull request by jonallured**
This PR adds structured data to the partner pages per the `LocalBusiness` schema.org spec:

https://schema.org/LocalBusiness

I poked around and discovered some things about the `address` and `location` attributes plus learned more about how we model partner locations so ended up pitching what you see here - we run both the `legalName` and `name` always and then conditionally run the `address` as a `PostalAddress` including phone number and conditionally run the `location` as a `Place` with lat/long in case Google will pick that up and do a map. More docs for the curious:

https://schema.org/Place
https://schema.org/PostalAddress

Note that I discovered there was no need to use the scary `dangerouslySetInnerHTML` api and instead I just ran the content right inside the `script` tag.

Looks like this:

Screen Shot 2021-08-06 at 10 55 31 AM

And here's the validator output:

Screen Shot 2021-08-06 at 11 01 30 AM

## Extract Stand-alone Structured Data Component

My first step here was to read the existing code for running structured data. I felt there was room for improvement there so I started this line of work by extracting a component called `StructuredData` that does the work of converting a JS object to stringified data and sets the correct context. It also encapsulates the funky `application/ld+json` content type and knows to run it as a `script` tag. I got some tests around this too.

## Migrate to Extracted Component

Once I had my extracted component I could then look at the places were I wanted to migrate away from the repetition. I started with the artist one and then did the artwork ones. Added tests here too. Easy peasy.

That left the breadcrumb one and that took a bit more doing because there was actual (untested?) behavior. So I took a few commits to extract and wrap with tests to ensure I could migrate without any regressions.

## Partner Page

At this point I was ready to add my new `LocalBusiness` component following these new patterns and then update the partner app to use it. One thing I struggled with a bit was making it so that `undefined` properties didn't come through and end up rendering to the page. I found that [Underscore `pick`](https://underscorejs.org/#pick) worked so I pulled that in - open to other ideas here!

## Testing This Change

All we can really do here is copy/paste into the validator until this lands in production. From there we can wait for Google to crawl the site and report any errors in Google Search Console which I plan to do in a week or so.

https://artsyproduct.atlassian.net/browse/GRO-379

/cc @artsy/grow-devs

### CODE REVIEW

File: src/v2/Components/Seo/LocalBusiness.tsx
Comment (dzucconi) [inline src/v2/Components/Seo/LocalBusiness.tsx:3]: use lodash instead (we need a linter for this!)

Comment (dzucconi) [reply src/v2/Components/Seo/LocalBusiness.tsx:3]: also I'd just pull out the relevant functions instead of importing _

File: src/v2/Components/Seo/LocalBusiness.tsx
Comment (dzucconi) [inline src/v2/Components/Seo/LocalBusiness.tsx:13]: I know we can import a package of these types instead of manually doing them. :looking:

Comment (dzucconi) [reply src/v2/Components/Seo/LocalBusiness.tsx:13]: https://www.npmjs.com/package/schema-dts

File: src/v2/Components/Seo/LocalBusiness.tsx
Comment (jonallured) [reply src/v2/Components/Seo/LocalBusiness.tsx:3]: Damn I def searched the package.json file for `lowdash` and not finding it searched for `underscore` found it and picked that instead, haha. Will update!

Comment (jonallured) [reply src/v2/Components/Seo/LocalBusiness.tsx:3]: Also confused by this feedback @dzucconi 🤔 

Screen Shot 2021-08-06 at 11 26 18 AM

Comment (dzucconi) [reply src/v2/Components/Seo/LocalBusiness.tsx:3]: Size concerns aside: we don't use underscore at all in the new app so importing it a single time will add the entire library unnecessarily. Theoretically lodash can be tree-shaken (hence advice importing just the functions you need) though I'm unclear as to whether or not we're reaping the benefit of that.

Comment (dzucconi) [reply src/v2/Components/Seo/LocalBusiness.tsx:3]: `lowdash` !== `lodash`

File: src/v2/Components/Seo/LocalBusiness.tsx
Comment (dzucconi) [reply src/v2/Components/Seo/LocalBusiness.tsx:13]: I've used this before and it really takes the guess work out of it, and types don't add any weight to the bundle since they are discarded

Comment (jonallured) [reply src/v2/Components/Seo/LocalBusiness.tsx:13]: Nice one, will add! 🙌

File: src/v2/Components/Seo/LocalBusiness.tsx
Comment (jonallured) [reply src/v2/Components/Seo/LocalBusiness.tsx]: Fixed with d7b13a1 - is that what you were thinking?

Comment (dzucconi) [reply src/v2/Components/Seo/LocalBusiness.tsx]: Exactly 👍

File: src/v2/Apps/Partner/Components/PartnerMeta/index.tsx
Comment (damassi) [inline src/v2/Apps/Partner/Components/PartnerMeta/index.tsx:16]: as a helpful nudge, can we remove these as we come across them? Mostly just adding safe access operators (or guards)

File: src/v2/Apps/Partner/Components/PartnerMeta/index.tsx
Comment (damassi) [inline src/v2/Apps/Partner/Components/PartnerMeta/index.tsx:34]: can use the `extractNodes` helper

File: src/v2/Components/Seo/StructuredData.tsx
Comment (damassi) [inline src/v2/Components/Seo/StructuredData.tsx:13]: nice

**Review by damassi — APPROVED** [review]
Code and tests look great and I'll leave it to you to merge and follow-up with @dzucconi's lib suggestion, or hold this until that's in 👍

File: src/v2/Components/Seo/LocalBusiness.tsx
Comment (jonallured) [reply src/v2/Components/Seo/LocalBusiness.tsx:13]: More chatter here but I had trouble adding this package and then using it for the `PostalAddress` data:

https://artsy.slack.com/archives/CP9P4KR35/p1628277103099800

Going to bail on this and can return to this topic later if it's warranted - hope that's ok!

File: src/v2/Apps/Partner/Components/PartnerMeta/index.tsx
Comment (jonallured) [reply src/v2/Apps/Partner/Components/PartnerMeta/index.tsx]: Took a shot at this with 620e78d - is that what you had in mind?

File: src/v2/Apps/Partner/Components/PartnerMeta/index.tsx
Comment (jonallured) [reply src/v2/Apps/Partner/Components/PartnerMeta/index.tsx]: Did this with fbc5cc4.

**Comment by jonallured** [comment]
Going to merge this, thanks for the feedback all!! 🤗

File: src/v2/Components/Seo/LocalBusiness.tsx
Comment (dzucconi) [reply src/v2/Components/Seo/LocalBusiness.tsx:13]: Yah is fine. FWIW looking at this now — you wouldn't use `PostalAddress` directly but rather the entire `LocalBusiness` type. If we ever want to do JSON+LD in the future we should definitely just use the provided schema. I suspect that these will be invalid when there's some nulls around, etc.

**Pull request merged**
